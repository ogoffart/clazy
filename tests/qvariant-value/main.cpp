#include <QtCore/QVariant>
#include <QtCore/QUrl>


enum Foo {
    Foo1
};
Q_DECLARE_METATYPE(Foo)


void test()
{
    QVariant v;
    v.toUrl(); //ok
    v.toBool(); //ok
    v.value<QUrl>(); //warn
    v.value<bool>(); //warn
    v.value<float>(); //warn
    v.value<int>(); //warn
    v.value<Foo>(); //ok
    QVariant(45).value<long int>();
}
