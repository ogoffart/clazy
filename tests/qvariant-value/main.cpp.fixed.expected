#include <QtCore/QVariant>
#include <QtCore/QUrl>


enum Foo {
    Foo1
};
Q_DECLARE_METATYPE(Foo)


void test()
{
    QVariant v;
    v.toUrl(); //ok
    v.toBool(); //ok
    qvariant_cast<QUrl>(v); //warn
    qvariant_cast<bool>(v); //warn
    qvariant_cast<float>(v); //warn
    qvariant_cast<int>(v); //warn
    qvariant_cast<Foo>(v); //ok
    qvariant_cast<long int>(QVariant(45));
}
